test/EVMHost.cpp

```diff
@@ -295,35 +303,32 @@ evmc::result EVMHost::call(evmc_message const& _message) noexcept
 			keccak256(bytes(message.input_data, message.input_data + message.input_size)).asBytes()
 		), h160::AlignRight);
 
-		message.destination = convertToEVMC(createAddress);
-		if (accounts.count(message.destination) && (
-			accounts[message.destination].nonce > 0 ||
-			!accounts[message.destination].code.empty()
+		message.recipient = convertToEVMC(createAddress);
+		if (accounts.count(message.recipient) && (
+			accounts[message.recipient].nonce > 0 ||
+			!accounts[message.recipient].code.empty()
 		))
 		{
-			evmc::result result({});
+			evmc::Result result;
 			result.status_code = EVMC_OUT_OF_GAS;
 			accounts = stateBackup;
 			return result;
 		}
 
 		code = evmc::bytes(message.input_data, message.input_data + message.input_size);
 	}
-	else if (message.kind == EVMC_DELEGATECALL || message.kind == EVMC_CALLCODE)
```

Comment:
This is an important change, in the new API there's a distinction between `recipient` and `code_address`, no need to manually track it.